### PR TITLE
Remove vendored `pulumigrpc.ResourceProviderServer` implementation

### PR DESCRIPTION
**Pull request by iwahbe**
As part of https://github.com/pulumi/pulumi-terraform-bridge/issues/2148, this PR removes the duplicate implementation of `pulumigrpc.ResourceProviderServer` copied over from pulumi/pulumi and then modified.

### CODE REVIEW

File: pf/internal/plugin/provider_server.go [hunk 1]
```diff
@@ -16,702 +16,80 @@ package plugin
...
-	return nil, status.Error(codes.Unimplemented, "Attach is not yet implemented")
+	p := &providerServer{
+		ResourceProviderServer: plugin.NewProviderServer(
+			providerThunk{
+				configencoding.New(configEncoding, NewProvider(provider)),
+			},
+		),
+	}
+
+	// This is a hack to work around limitations in the Pulumi protocol handshake:
```
Comment (VenelinMartinov) [inline pf/internal/plugin/provider_server.go:40]: Is this hack new? We need to add tests for this if so.

**Review by VenelinMartinov — COMMENTED** [review]
Great that we're removing code copied from pu/pu but I'm a bit confused around the context of this - is the idea that the configencoding provider will be the default pf provider?

Also what is the state of the pkg/ config encoding? Is this used currently? The changes there look non-trivial, is this tested?

**Comment by t0yv0** [comment]
I remember some unfortunate history here where we started vendoring code from p/p and adding Context, but continued editing it in-place. The full story is here:

https://github.com/pulumi/pulumi-terraform-bridge/commits/master/pf/internal/plugin/provider_server.go

The suspect PRs that required edits are:

https://github.com/pulumi/pulumi-terraform-bridge/pull/1716 Attach
https://github.com/pulumi/pulumi-terraform-bridge/pull/1683 DiffConfig
https://github.com/pulumi/pulumi-terraform-bridge/pull/1047 Do not accept secrets
https://github.com/pulumi/pulumi-terraform-bridge/pull/1065 Enable non-string config values for Plugin Framework

File: pf/internal/plugin/provider_server.go [hunk 2]
```diff
@@ -16,702 +16,80 @@ package plugin
...
+	//
+	// We want to be able to return secret values from CheckConfig, but we don't know
+	// if the engine accepts secrets until Configure is called (after
+	// CheckConfig). [plugin.NewProviderServer] defaults to not sending secrets, which
+	// is safe but breaks existing behavior.
+	//
+	// We send an artificial Configure call to tell the
+	// [pulumirpc.ResourceProviderServer] that it should accept
+	// secrets. [providerThunk] acts as a thunk, preventing the underlying
+	// [ProviderWithContext] from witnessing the extra Configure call.
```
Comment (t0yv0) [inline pf/internal/plugin/provider_server.go:50]: What is the existing behavior? Are bridged providers assuming that the engine accepts secrets, even though the older versions of the engine do not?

File: pkg/tfbridge/config_encoding.go [hunk 3]
```diff
@@ -121,126 +113,90 @@ func (*ConfigEncoding) zeroValue(typ shim.ValueType) resource.PropertyValue {
...
-	if pv.IsString() {
-		jsonString = pv.StringValue()
-		jsonStringDetected = true
-	}
+func (enc *ConfigEncoding) UnfoldProperties(m resource.PropertyMap) (resource.PropertyMap, error) {
+	result := make(resource.PropertyMap, len(m))
+	for _, pk := range m.StableKeys() {
+		v := m[pk]
+
+		// If we don't have type information on the field, we don't attempt to do
```
Comment (t0yv0) [inline pkg/tfbridge/config_encoding.go:135]: There's too many changes to config_encoding for this PR. There's semantic changes here it looks like, adding to the unease. I'd like to request making these changes step-by-step for easier review to understand why this is happening, so we can double-check this is being tested properly and makes sense.

There is this PR by @blampe 
https://github.com/pulumi/pulumi/pull/15032 that would still be good to land to not use this encoding at all. 

I'm trying to recall the rollut plan there. Once the bridge makes sure that fix is rolled out to the code generators, subsequent versions might not ever need to deal with this encoding at all and may drop the code cold-turkey.

**Review by t0yv0 — CHANGES_REQUESTED** [review]
Looks a bit risky to no end-user benefit, suggest breaking down to derisk.

File: pf/internal/plugin/provider_server.go (hunk 2 above)
Comment (t0yv0) [reply pf/internal/plugin/provider_server.go:50]: Actually, curious, can we fix the protocol and send this information (whether the engine supports secrets or not) as part of CheckRequest.

Comment (iwahbe) [reply pf/internal/plugin/provider_server.go:50]: If we are going to change the protocol:

We shouldn't do it as part of `CheckRequest`, we should have the engine send a `HandshakeRequest` that handles protocol level negotiation. Let's leave `CheckConfig` and `Configure` to be provider level options.

File: pf/internal/plugin/provider_server.go (hunk 1 above)
Comment (iwahbe) [reply pf/internal/plugin/provider_server.go:40]: The hack is new but the behavior is already under test. Removing the hack fails tests.

**Pull request closed without merging**